plugins/module_utils/data_set.py

```diff
@@ -1717,6 +1797,69 @@ def _process_listcat_output(self, output):
         return result
 
 
+class MVSDataSet():
```

Comment:
i'm curious as to the purpose of a new `MVSDataSet` class. Primarily, I'm unclear on why the existing DataSet class cant be enhanced. 

Also, thinking into future maintainability, will it create confusion if there's a DataSet object going around as well as an MVSDataSet object? Will it be clear which class/type is best suited for any given scenario?

Reply:
Sorry I left that part out, I was in a rush and didn't explain a lot of things in this PR.

So, DataSet class description is `Perform various data set operations such as creation, deletion and cataloging.`, in a sense is more like a DatasetUtils module, we do have plenty of static methods effectively using this class as a library.

New proposed MVSDataSet class is meant to be a representation of a single data set during the execution of a module, took this one from my current work in `zos_data_set` module, by having a single dataset attributes encapsulated in a class we can pass around the object instead of having functions with many arguments in their signature.

Effectively changing it from this :

<img width="406" alt="Screenshot 2024-05-02 at 9 33 57 AM" src="https://github.com/ansible-collections/ibm_zos_core/assets/23223623/aabbd099-698d-446b-9297-8518b5f3778c">


To this (still in progress) :

<img width="176" alt="Screenshot 2024-05-02 at 9 34 50 AM" src="https://github.com/ansible-collections/ibm_zos_core/assets/23223623/22045019-c367-40e0-a5fc-195fed241cf4">

Apart from reducing the number of arguments encapsulation benefits us when having to do data set operations like creating a new dataset or copying it, by having its own attributes in the object and not in some other variables scattered around we can simplify complex flows like in `zos_copy`.

Another benefit is that this object at the moment of initialization has attributes `raw_name` and `name`, `name` is escaped in case of needing it and if the name is a gds relative name (non positive one) is resolved into an absolute name, meaning, that `name` contains the name we use internally for the data set operations, whilst `raw_name` is the original name, so that when we need to show any error message to our users we display `data_set.raw_name`.

MVSDataSet was used because DataSet was already taken and updating it would have to be done in 120+ places. I'm open for suggestions just added this class into this PR because Ivan, Rich and Marcel have work items that depend on this, another option name was zDataSet...

Reply:
that makes sense to me! thanks for the explanation